core/src/main/scala/org/apache/spark/scheduler/dynalloc/ExecutorMonitor.scala

```diff
@@ -550,7 +550,7 @@ private[spark] class ExecutorMonitor(
 
     // The set of shuffles for which shuffle data is held by the executor.
     // This should only be used in the event thread.
-    private val shuffleIds = if (shuffleTrackingEnabled) new mutable.HashSet[Int]() else null
+    private val shuffleIds = new mutable.HashSet[Int]()
```

Comment:
Please revert this part because `if (shuffleIds != null && shuffleIds.nonEmpty)` is totally fine.
For this kind of PR, we need to minimize as much as possible in order to backport easily, @warrenzhu25 .

Reply:
Done.